sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -377,6 +379,18 @@ public ConfigurationClientBuilder serviceVersion(ConfigurationServiceVersion ver
         return this;
     }
 
+    /**
+     * Sets the customized headers that is used when each request is sent. It is possible to use new header value for
+     * each request.
+     *
+     * @param headersSupplier user's headers supplier applied to each request.
+     * @return The updated ConfigurationClientBuilder object.
+     */
+    public ConfigurationClientBuilder headersSupplier(Supplier<HttpHeaders> headersSupplier) {
+        this.headersSupplier = headersSupplier;
```

Comment:
The `HttpPipelinePolicy` is configured to throw an exception if the supplier is null, if you are going to add a `CustomHeaderPolicy` into the pipeline within the builder you'll need to ensure this is not null.